external-api/src/v1/controllers/number-generator.controller.ts

```diff
@@ -1,85 +1,92 @@
-// Number Generator API is used to get a new deal/facility id.
-// However, the id might already be in use so we need to check it with ACBS.
-//
-// the flow is:
-// 1) Call number generator Azure function
-// 2) return a PENDING status while Azure function runs independently
-// 3) Create a scheduled job to keep checking the Azure function status
-
-import axios from 'axios';
 import * as dotenv from 'dotenv';
+import axios, { HttpStatusCode } from 'axios';
 import { Request, Response } from 'express';
-import * as durableFunctionsLogController from './durable-functions-log.controller';
-
+import { NumberGeneratorResponse, NumberGeneratorErrorResponse } from '../../interfaces';
+import { InvalidEntityTypeError } from '../errors';
+import { ENTITY_TYPE, NUMBER_TYPE, USER } from '../../constants';
 dotenv.config();
-const numberGeneratorFunctionUrl = process.env.AZURE_NUMBER_GENERATOR_FUNCTION_URL;
 
-const callNumberGenerator = async ({ dealType, entityType, entityId, dealId, user }: any) => {
-  const response: any = await axios({
-    method: 'POST',
-    url: `${numberGeneratorFunctionUrl}/api/orchestrators/numbergenerator`,
-    data: {
-      dealType,
-      entityType,
-      entityId,
-      dealId,
-      user,
-    },
-  }).catch((error: any) => ({
-    error,
-  }));
+const { APIM_MDM_URL, APIM_MDM_VALUE, APIM_MDM_KEY } = process.env;
 
-  if (response.error) {
-    await durableFunctionsLogController.addDurableFunctionLog({
-      type: 'NUMBER_GENERATOR',
-      data: {
-        dealType,
-        entityType,
-        entityId,
-        dealId,
-        user,
-        error: await response.error.toJSON(),
-      },
-    });
+const headers = {
+  headers: {
+    [String(APIM_MDM_KEY)]: APIM_MDM_VALUE,
```

Comment:
Is this some kind of API key? values aren't very clear here to know what this is adding - also surprised we're putting it in an array

Reply:
Correct for APIM, this is to keep its value not an array.